### PR TITLE
Regular app audit

### PR DESCRIPTION
**Pull request by mwbrooks**
### Type of change (place an x in the [ ] that applies)

- [ ] New sample
- [ ] New feature
- [x] Bug fix
- [ ] Documentation

### Summary

This pull request audits the sample app and makes the following changes:

* Update `.gitignore` to only ignore the local apps
* Add a default `.slack/apps.json` with the JSON content `{}`
* Removes the `deno.lock`
* Disables the deno lock file
  * Group discussion decided to disable it and allow developers to opt-in to using it
* Minor comment tweaks
* Fixes a date format issue introduced by the latest Deno runtime (v1.33.x)
  * Before: `new Date().toLocaleDateString("en-CA")` returns `YYYY-MM-DD`
  * Now: `new Date().toLocaleDateString("en-CA")` returns `mm/dd/YYYY`
  * This caused a runtime validation error when the Workflow was started
  * Thankfully, the date format seems to work fine for the datastore expressions

Tested functionality:
* [x] Linting & Tests pass
* [x] Logged a run
* [x] Displayed the leaderboard
* [x] Scheduled the displaying of the leaderboard

### Requirements (place an x in each [ ] that applies)

- [x] I’ve checked my submission against the Samples Checklist to ensure it complies with all standards
- [x] I have ensured the changes I am contributing align with existing patterns and have tested and linted my code
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct)

### CODE REVIEW

File: .slack/apps.json
Comment (e-zim) [inline .slack/apps.json:1]: Nice catch! 👏

File: deno.jsonc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "importMap": "import_map.json",
+  "lock": false,
```
Comment (e-zim) [inline deno.jsonc:3]: 🔒 👍

**Review by e-zim — CHANGES_REQUESTED** [review]
Appreciate the polishing here! With the change to how dates are being formatted, there might some updates needed to the datastore query too. At the moment, it seems like the following expression might not be collecting the logged values:

https://github.com/slack-samples/deno-virtual-running-buddies/blob/67af236e2352095841c36077d47a6136f7d7847f/functions/collect_team_stats.ts#L69-L79

There could also be an oddity with the `BETWEEN` statement, unrelated to these changes, but I'm finding that my most entry isn't being collected in this query:

dates

If you think this might be unrelated to this PR, I'm happy to investigate this in another pull, just let me know!

**Comment by e-zim** [comment]
>   * Before: new Date().toLocaleDateString("en-CA") returns YYYY-MM-DD
>   * Now: new Date().toLocaleDateString("en-CA") returns mm/dd/YYYY

IMO, we might want to standardize around the (wonderful) ISO 8601 format for dates if we're making these changes. mm/dd/YYYY seems to be in a confusing order?

**Review by filmaj — APPROVED** [review]
Yes, this is good, and fixes #12, so I'm going to merge this.

As for the issue you identified @e-zim , I can reproduce this too. I'm going to file a separate issue for that and look into it a bit more and we can address in a separate PR.

**Pull request merged**